### PR TITLE
doc to show how to use broker via vendor

### PR DESCRIPTION
**Pull request by jmrodri**

### CODE REVIEW

File: docs/using_as_library.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
+# Using as a library
+
+The Automation Broker implements the Open Service Broker API. As a broker it
+provides quite a bit of behavior that might want to be used by other broker
+authors, for example, asynchronous job support, ...
```
Comment (djzager) [inline docs/using_as_library.md]: nit: `asynchronous job support, etc.`

File: docs/using_as_library.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+## Vendoring
+
+To get the Automation Broker as a dependency, here is an example of the
+`Gopkg.toml` used by `dep`. You can use any of the types supported by the `dep`
```
Comment (djzager) [inline docs/using_as_library.md]: Here is an excerpt from `Gopkg.toml` used by [golang/dep](https://golang.github.io/dep/) that adds the Automation Broker as a dependency.

File: docs/using_as_library.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+```toml
+[[constraint]]
+  branch = "master"
+  name = "github.com/openshift/ansible-service-broker"
+```
+
+## Extending
+
+Once we have the dependency installed in our project, how do we use it? The
+Automation Broker will do most if not all of the heavy lifting, but allows the
```
Comment (djzager) [inline docs/using_as_library.md]: nit: will do most, if not all, ...

File: docs/using_as_library.md (hunk 1 above)
Comment (jmrodri) [reply docs/using_as_library.md]: addressed

File: docs/using_as_library.md (hunk 2 above)
Comment (jmrodri) [reply docs/using_as_library.md]: addressed

File: docs/using_as_library.md (hunk 3 above)
Comment (jmrodri) [reply docs/using_as_library.md]: addressed

**Pull request merged**